[merged] Add URL override for fetching objects

krnowak opened the pull request:
This adds an `override-url` option to `ostree_repo_pull_with_options`, which, if specified, overrides the URL taken from repo config for a given remote.

This also adds a `--url` option to `ostree pull` which uses the above option.

cgwalters commented:
Can you add a test case?  Examples are in `tests/test-pull-commit-only.sh` and `tests/pull-test.sh` etc.

cgwalters commented:
One test case would be to have a repo that has an upstream remote, clone the remote into a different path, then rm -rf the original, then do a pull with `--url` from the mirror.

krnowak commented:
Sure, will get to it tomorrow.

krnowak commented:
Updated. I added a `--log-file` flag to trivial-httpd, because I couldn't get why the test was failing. Apparently I did not pass the `--mirror` flag to `ostree pull`, so fetching `refs/heads/main` from the mirror was returning 404. Not sure if you want it in a separate PR, please let me know.

cgwalters commented on src/ostree/ot-builtin-trivial-httpd.c:
```diff
@@ -351,6 +385,30 @@ ostree_builtin_trivial_httpd (int argc, char **argv, GCancellable *cancellable,
+    {
+      GOutputStream *stream = NULL;
+
+      if (g_strcmp0(opt_log, "-") == 0)
```
Missing space between identifier and paren here.

cgwalters commented:
Test LGTM.

krnowak commented:
Updated. Added the check for `--daemonize` and `--log-file=-` too.

cgwalters commented:
@cgwalters-bot r+ 27fa1ac

cgwalters commented:
In the future, I generally prefer test cases to come in the same commit as the code introducing them.  Not a big deal though.

krnowak commented:
Alright, will remember next time.
